### ISSUE TITLE
Floating point numbers: Clarification

### ISSUE THREAD

**Issue by dbrgn** [fontobene/fontobene#42]
We write:

> The coordinates are arbitrary precision floating point numbers using the full stop (U+002E) as decimal mark

...but there's no specification whether exponential notation, NaN or Infinity are valid values. Do we need to explicitly exclude those, or should it be obvious?

**Comment by rnestler** [comment]
> ...but there's no specification whether exponential notation, NaN or Infinity are valid values. Do we need to explicitly exclude those, or should it be obvious?

What would be the purpose of NaN and infinity? Defining a curve if infinite curvature which will then be a straight line :wink:

**Comment by dbrgn** [comment]
Nothing, but we should probably explicitly exclude them from the valid values :slightly_smiling_face: It's relevant to parsers, even if the meaning would be nonsense.

**Comment by ubruhin** [comment]
I would prefer to explicitly forbid exponential notation etc. in the specs, resp. define a regex or so how a valid number has to look like.

**Comment by rnestler** [comment]
> I would prefer to explicitly forbid exponential notation etc. in the specs, resp. define a regex or so how a valid number has to look like.

Why would you forbid exponential notation? It is quite useful for small or big numbers. Or do values larger than 1000 or smaller than 0.001 almost never occur?

**Comment by dbrgn** [comment]
> Or do values larger than 1000 or smaller than 0.001 almost never occur?

Yes, read the spec :wink: https://github.com/fontobene/fontobene/blob/master/SPECIFICATION.md#coordinate-system
